updated w/ resolved changes - Feature/kinematics drive

johnylamw opened the pull request.

johnylamw committed:
- updated PoseSubsystem with override of setCurrentPosition and better doubleproperty and function naming
- integrated kinematicsDrive
- arcade drive with kinematics drive - also added functions used for tuning
- new deadband logic (not sure if it will be commited)

johnylamw committed:
- updated PoseSubsystem with override of setCurrentPosition and better doubleproperty and function naming
- integrated kinematicsDrive
- arcade drive with kinematics drive - also added functions used for tuning
- new deadband logic (not sure if it will be commited)

johnylamw committed:
Merge branch 'master' into feature/KinematicsDrive

JohnGilb commented on src/main/java/competition/subsystems/drive/commands/ArcadeDriveWithKinematicsCommand.java:
```diff
@@ -0,0 +1,82 @@
+    final DoubleProperty linearVelocityConstantValue;
+    final DoubleProperty angularVelocityConstantValue;
+
+    double linearVelocityConstant = 1; // needs tuning
```
you can delete these now that you have the properties above

JohnGilb commented on src/main/java/competition/subsystems/drive/commands/ArcadeDriveWithKinematicsCommand.java:
```diff
@@ -0,0 +1,82 @@
+        this.oi = oi;
+        this.addRequirements(drive);
+
+        linearVelocityConstantValue = propManager.createEphemeralProperty("Linear Velocity Constant Value", 0.0);
```
these need to be persistent (e.g. `propManager.createPersistentProperty`) so that they will stick around once we get good values.

JohnGilb commented on src/main/java/competition/subsystems/drive/commands/ArcadeDriveWithKinematicsCommand.java:
```diff
@@ -0,0 +1,82 @@
+        this.addRequirements(drive);
+
+        linearVelocityConstantValue = propManager.createEphemeralProperty("Linear Velocity Constant Value", 0.0);
+        angularVelocityConstantValue = propManager.createEphemeralProperty("Angular Velocity Constant Value", 0.0);
```
you should be able to guesstimate values for these constants. These essentially are "max speeds" in a way, so what is a reasonable max speed for our robot?

JohnGilb commented on src/main/java/competition/subsystems/drive/commands/ArcadeDriveWithKinematicsCommand.java:
```diff
@@ -0,0 +1,82 @@
+    }
+
+    // Functions for robot tuning
+    public void increaseLinearVelocityConstant(double increaseRate) {
```
You don't need methods for this - the joy of DoubleProperties is that they can be changed immediately on the Driver Station using the Shuffleboard instead of writing code like this

JohnGilb reviewed (changes requested):
Also, tests!

johnylamw committed:
- DriveSubSystem: added pre-set ks/kv & trackwidth values in order to test if the kinematics system work. would need characterization tool to get the proper values

- ArcadeDriveWithKinematicsCommand: removed tuning functions since we can use persistent double property and change the values in shuffleboard

- KinematicsDriveTest: a drive test for the kinematicsdrive function

johnylamw committed:
Merge branch 'master' into feature/KinematicsDrive

johnylamw committed:
-removed unused imports

johnylamw committed:
Merge branch 'master' into feature/KinematicsDrive
